pallets/nfts/src/lib.rs

```diff
@@ -326,45 +286,37 @@ impl<T: Config> NFTs for Pallet<T> {
     type NFTDetails = NFTDetails;
     type NFTId = T::NFTId;
     type NFTSeriesId = NFTSeriesId;
+    type Protocol = Protocol;
 
     fn create(
         owner: &Self::AccountId,
         details: Self::NFTDetails,
     ) -> result::Result<Self::NFTId, DispatchError> {
-        // Check if the owner is even allowed to add anything to the series.
-        let mut nft_series = sp_std::vec![];
+        // Check for capsule prerequisites.
+        if details.is_capsule {
+            Self::check_capsule_prerequisites(owner, details.protocol.clone())?;
+        }
+
+        // Check for series prerequisites.
         let series_id = details.series_id;
+        let mut nft_series = Self::get_nft_series_data(owner, series_id)?;
 
-        if series_id != NFTSeriesId::default() {
-            if let Some(series) = Series::<T>::get(series_id) {
-                ensure!(series.owner == *owner, Error::<T>::NotSeriesOwner);
-                nft_series = series.nfts;
-            }
-        }
+        // Get current and next nft id.
+        let (nft_id, next_nft_id) = Self::get_next_nft_id()?;
 
-        // Create the nft.
-        let nft_id = Total::<T>::get();
-        Total::<T>::put(
-            nft_id
-                .checked_add(&1.into())
-                .ok_or(Error::<T>::NFTIdOverflow)?,
-        );
+        if details.is_capsule {
+            Self::pay_fees(owner)?;
+        }
```

Comment:
I already had modified the pallet to charge a minting fee. Probably do not want to do this for now. We will have separate functions once we integrate with Intel SGX too.

Reply:
I just moved your code around. AFAIK, the mint fee should only be paid if you want to create a capsule. 
If you meant that paying fees is exclusive to the `create` extrinsic and that it should not occur when calling the trait function, then I will of course reverse my changes.

Reply:
Let's keep where it was. It would make sense to charge a fee for NFT minting for now

Reply:
Sure :)